### PR TITLE
Deploy several documentations

### LINKED ISSUE #693: Make sure that the documentation on the website is that of the last release
Not of unreleased code ;)

### PR DESCRIPTION
**Pull request by Halbaroth**
This PR adds the ability to deploy several documentations of Alt-Ergo. The new documentation workflows work as follows:
- the workflow `build_doc` builds both sphinx and odoc documentations and merges them in a single artifact. This workflow is trigger after pushing on PR;
- while pushing on `next`, the new `gh-pages` workflow retrieves the previous artifact and deploy it in the directory `/dev` of the `gh-pages` branch;
- if we want to add a new release documentation, we have to do by hand using the target `doc` of our makefile and update the `index.mld` file of `gh-pages`.

This PR fixes #693 

I also removed the support for `gtk-lang` of the native input because we removed the GUI.

### CODE REVIEW

File: .github/workflows/build_doc.yml [hunk 1]
```diff
@@ -53,9 +53,7 @@ jobs:
...
       - name: Deploy
-        uses: peaceiris/actions-gh-pages@v4
+        uses: JamesIves/github-pages-deploy-action@v4
         with:
-          deploy_key: ${{ secrets.ACTIONS_DEPLOY_KEY }}
-          publish_dir: build_doc
-          destination_dir: dev
-          enable_jekyll: true
+          folder: build_doc
+          target-folder: dev
```
Comment (bclement-ocp) [inline .github/workflows/build_doc.yml]: We used to have:

```
      # Deploy files contain in _build directory on gh-pages branch
      - name: deploy-doc
        uses: JamesIves/github-pages-deploy-action@3.6.2
        with:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          BRANCH: gh-pages
          FOLDER: _build
          CLEAN: false
```

does that no longer work?

Comment (bclement-ocp) [reply .github/workflows/build_doc.yml]: Also possibly you need to deploy from a branch in the repository (not from a fork)?

Comment (bclement-ocp) [reply .github/workflows/build_doc.yml]: You can try directly on your fork if you enable GH actions on your fork (although you may need to enable GH pages as well, not sure)

Comment (Halbaroth) [reply .github/workflows/build_doc.yml]: Both does not work so it seems we cannot deploy from a PR I guess. I will try on my fork.

Comment (Halbaroth) [reply .github/workflows/build_doc.yml]: It works on my fork... aahhhhhhhhhhhhhhhh

**Comment by Halbaroth** [comment]
I gave up.

**Pull request closed without merging**